Add Kong Gateway Operator docs (#6144)
Kong/docs.konghq.com: app/_src/gateway-operator/get-started/konnect/create-route.md

---
title: Create a Route
content-type: tutorial
book: kgo-konnect-get-started
chapter: 3
---

{:.note}

> > The following examples require you to have `jq` installed on your machine. You can use the [Konnect UI](https://cloud.konghq.com) to configure services and routes if you cannot install `jq`.

Reviewer: ```suggestion
> These examples require that you have `jq` installed on your machine. You can also use the [Konnect UI](https://cloud.konghq.com) to configure services and routes.
```

> Your `DataPlane` should now be running, but there are no routing rules to tell {{ site.base_gateway }} how to proxy traffic. In this tutorial we will use the Konnect API to configure and test routing.

Reviewer: ```suggestion
After you deploy the`DataPlane` node it is  connecetd to the your cluster plane, but there are no routing rules  for {{ site.base_gateway }} to proxy traffic. You can use Konnect API to configure and test routing.
```

Author: Neither option feels right here. How about:

```suggestion
Once your `DataPlane` is deployed it will receive configuration from Konnect, but there are no routing rules available for {{ site.base_gateway }} to proxy traffic. You can use Konnect API to configure and test routing.
```

Reviewer: > Neither option feels right here. How about:

How about:

After your 'DataPlane`is deployed it receives configuration from Konnect, but there are no routing rules available for {{ site.base_gateway }} to proxy traffic. You can use Konnect API to configure and test the routing.

> ## Prerequisites

Reviewer: ```suggestion
## Before you begin
```

> ## Prerequisities

Reviewer 2: ```suggestion
## Prerequisites
```

> 1. Visit the Gateway Manager section of Konnect and fetch your control plane ID. This is available on the _Data Plane Nodes_ page next to the _Control Plane ID_ label. Export a variable that contains this value:

Reviewer: 1. In Gateway Manager, navigate to the **Overview** page of the control plane to which you added the data plane.
1. Copy the ID of the control plane in the **About this Hybrid Control Plane** section.
1. Set the value of the variable `CP_UUID` to the ID of the control plane that you copied.

    ```bash
    export CP_UUID="a62e9a4c-d47..."
    ```

> 1. Generate a [personal access token](https://cloud.konghq.com/global/account/tokens) to use the API. Export this as `KONNECT_TOKEN` so that we can authenticate with the Konnect API:

Reviewer: ```suggestion
1. Generate a [personal access token](https://cloud.konghq.com/global/account/tokens) to use the Konnect API.
1. Set the value of the variable `KONNECT_TOKEN` to the personal access token that you generated.
```

    ```bash
    export KONNECT_TOKEN="kpat_pYyaCgDWv..."
    ```

## Create Kong Entities

> Once the above variables are set, you can use them to create a service and a route in your {{ site.konnect_short_name }} control plane.

Reviewer: ```suggestion
1.  Create a service and a route in your {{ site.konnect_short_name }} control plane using the variables that you set.
```

The following commands create a service named _Sample_ and a route that proxies all traffic to mockbin.org:

```bash
SERVICE_ID=$(curl -sS -H "Authorization: Bearer $KONNECT_TOKEN" https://us.api.konghq.com/v2/runtime-groups/$CP_UUID/core-entities/services -d name=Sample -d url=https://mockbin.org | jq -r .id)

curl -sS -H "Authorization: Bearer $KONNECT_TOKEN" https://us.api.konghq.com/v2/runtime-groups/$CP_UUID/core-entities/services/$SERVICE_ID/routes -d "paths[]=/"
```

## Send test traffic

> Once the service and route are created, you can send traffic to the proxy and it will forward the request to mockbin.org. We'll use Mockbin's `/request` endpoint to echo back the request we made in the response.

Reviewer: ```suggestion
After the service and route are created, you can send traffic to the proxy and it forwards the request to mockbin.org. You can use Mockbin's `/request` endpoint to echo the request you make in the response.
```

To make a request to the proxy we need to fetch the LoadBalancer IP address using `kubectl get services`:

```bash
export PROXY_IP=$(kubectl get services -n kong -o json | jq -r '.items[] | .status.loadBalancer?|.ingress[]?|.ip')
echo "Proxy IP: $PROXY_IP"
```

{:.note}
> Note: if your cluster can not provision LoadBalancer type Services then you may not receive an IP address

> To test the routing rules that you just created, use `curl` to make a request to the proxy IP:

Reviewer: ```suggestion
1. Test the routing rules that you created send a request to the proxy IP address.
```

```bash
curl $PROXY_IP/request/hello
```

## Delete the test service

> You can delete the test service now that you've verified that {{ site.kgo_product_name }} and {{ site.konnect_short_name }} are working.

Reviewer: ```suggestion
You can delete the test service after you verify that {{ site.kgo_product_name }} and {{ site.konnect_short_name }} are working.
```

> Run the following to delete the `Sample` service:

Reviewer: ```suggestion
1.  To delete the `Sample` service.
```

```bash
curl -sS -H "Authorization: Bearer $KONNECT_TOKEN" https://us.api.konghq.com/v2/runtime-groups/$CP_UUID/core-entities/services/$SERVICE_ID -X DELETE
```

## Next steps

> Now that you have a running `DataPlane` that received configuration from Konnect you can explore all of the capabilities that {{ site.base_gateway }} provides.

Reviewer: ```suggestion
Now that you have a `DataPlane` that has configuration from Konnect you can explore all of the capabilities that {{ site.base_gateway }} provides.
```
